plugins/lookup/etcd_cluster_nodes.py

```
@@ -0,0 +1,96 @@
+
+        myvars = getattr(self._templar, '_available_variables', {})
+        # inventory hostname
+        ihn = variables['inventory_hostname']
+
+        # If no primary found in the inventory we return an empty list
+        if 'primary' not in variables['groups']:
+            return []
+
+        # # If no terms, we'll used the current hosts etcd_cluster_name if defined
+        if len(terms) == 0:
+            if 'etcd_cluster_name' in myvars['hostvars'][ihn]:
+                filter_etcd_cluster_name = myvars['hostvars'][ihn]['etcd_cluster_name']
+
+        if len(terms) > 0:
+            filter_etcd_cluster_name = terms[0]
+
+        # etcd can be deployed on BDR, barman and proxy nodes
+        for group in ( 'primary', 'standby', 'pemserver', 'pgbouncer', 'pgpool2',
+            'barmanserver', 'witness', 'proxy', 'pgbackrestserver', ):
+
+            if group not in variables['groups']:
+                continue
+
+            for host in variables['groups'][group]:
+                hostvars = myvars['hostvars'][host]
+
+                hv = myvars['hostvars'][host]
+
+                if 'etcd' not in hostvars:
+                    # Not a node being part of the etcd
+                    continue
+
+                if hostvars.get('etcd'):
+                    if 'etcd_cluster_name' not in hostvars:
+                        continue
+                    
+                    if hv['etcd_cluster_name'] != filter_etcd_cluster_name:
+                        continue
+
+                    etcd_node = dict(
+                        ansible_host=hostvars['ansible_host'],
+                        hostname=hostvars.get(
+                            'hostname', hostvars.get('ansible_hostname')
+                        ),
+                        private_ip=hostvars['private_ip'],
+                        inventory_hostname=hostvars['inventory_hostname'],
+                        etcd_cluster_name=hostvars.get('etcd_cluster_name', None),
```

This default of `None` is contradictory to the checks in lines 78-82. If there is no `etcd_cluster_name` set it would not reach this point. If the `etcd_cluster_name` is required, 78-79 should raise a failure, or else there should be some way to gather all etcd nodes without a defined cluster name.

---

Yes, good point. Following is fine:
```
etcd_cluster_name=hostvars['etcd_cluster_name'],
```

---

I see your point now. Good catch!